src/lifecycles/UiComponentLifecycle.ts

```diff
@@ -0,0 +1,65 @@
+/*
+Copyright 2023 Mikhail Aheichyk
+Copyright 2023 Nordeck IT + Consulting GmbH.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+export enum UiComponentLifecycle {
+    ShouldShowComponent = "should_show_component"
+}
+
+export type ShouldShowUiComponentOps = {
+    shouldShowComponent: boolean | undefined;
+};
+
+/**
+ * Listener for "should_show_component" request
+ */
+export type ShouldShowUiComponentListener = (
+    shouldShowUiComponentOps: ShouldShowUiComponentOps,
+    component: UiComponent,
+    matrixUserId: string
+) => void;
+
+export enum UiComponent {
```

Comment:
This enum is copied from https://github.com/nordeck/matrix-react-sdk/blob/209b65243a82d1131ce801e1f485c1488df04403/src/settings/UIFeature.ts#L37-L67 I am not sure what should be the correct approach to avoid duplicating this enum. Maybe move the enum into module api and to use in react sdk?